tests/integration-tests/tests/networking/test_on_demand_capacity_reservation.py

```diff
@@ -0,0 +1,167 @@
+    )
+    open_odcr = ec2.CapacityReservation(
+        "integTestsOpenOdcr",
+        AvailabilityZone=availability_zone,
+        InstanceCount=4,
+        InstancePlatform="Linux/UNIX",
+        InstanceType="c6gn.xlarge",
+    )
+    target_odcr = ec2.CapacityReservation(
+        "integTestsTargetOdcr",
+        AvailabilityZone=availability_zone,
+        InstanceCount=4,
+        InstancePlatform="Linux/UNIX",
+        InstanceType="c7g.xlarge",
+        InstanceMatchCriteria="targeted",
+    )
+    pg_name = placement_group_stack.cfn_resources["PlacementGroup"]
+    pg_odcr = ec2.CapacityReservation(
+        "integTestsPgOdcr",
+        AvailabilityZone=availability_zone,
+        InstanceCount=2,
+        InstancePlatform="Linux/UNIX",
+        InstanceType="m6g.xlarge",
+        InstanceMatchCriteria="targeted",
+        PlacementGroupArn=boto3.resource("ec2").PlacementGroup(pg_name).group_arn,
+    )
+    odcr_group = resourcegroups.Group(
+        "integTestsOdcrGroup",
+        Name=generate_stack_name("integ-tests-odcr-group", request.config.getoption("stackname_suffix")),
+        Configuration=[
+            resourcegroups.ConfigurationItem(Type="AWS::EC2::CapacityReservationPool"),
+            resourcegroups.ConfigurationItem(
+                Type="AWS::ResourceGroups::Generic",
+                Parameters=[
+                    resourcegroups.ConfigurationParameter(
+                        Name="allowed-resource-types", Values=["AWS::EC2::CapacityReservation"]
+                    )
+                ],
+            ),
+        ],
+        Resources=[
+            Sub(
+                "arn:${partition}:ec2:${region}:${account_id}:capacity-reservation/${odcr_id}",
+                partition=get_arn_partition(region),
+                region=region,
+                account_id=Ref("AWS::AccountId"),
+                odcr_id=Ref(open_odcr),
+            ),
+            Sub(
+                "arn:${partition}:ec2:${region}:${account_id}:capacity-reservation/${odcr_id}",
+                partition=get_arn_partition(region),
+                region=region,
+                account_id=Ref("AWS::AccountId"),
+                odcr_id=Ref(target_odcr),
+            ),
+            Sub(
+                "arn:${partition}:ec2:${region}:${account_id}:capacity-reservation/${odcr_id}",
+                partition=get_arn_partition(region),
+                region=region,
+                account_id=Ref("AWS::AccountId"),
+                odcr_id=Ref(pg_odcr),
+            ),
+        ],
+    )
+    odcr_template.add_resource(open_odcr)
+    odcr_template.add_resource(target_odcr)
+    odcr_template.add_resource(pg_odcr)
+    odcr_template.add_resource(odcr_group)
+
+    odcr_stack = CfnStack(
+        name=generate_stack_name("integ-tests-odcr", request.config.getoption("stackname_suffix")),
+        region=region,
+        template=odcr_template.to_json(),
+    )
+    cfn_stacks_factory.create_stack(odcr_stack)
```

Comment:
Can you refactor the code in to a fixture. Using fixture has the benefit of cleaning up resources even if the test fails. The current code does not clean up the ODCR if anything wrong before Line 157

Reply:
Sure, that was something I did initially but pulled back on due to the extra context to get the resource group.  I think it's a bit cleaner now, so I'll do that again!